examples/custom_column_resizing.html

```diff
@@ -0,0 +1,295 @@
+
+            getColumnKey(index) {
+                return this.columns[index].key;
+            }
+
+            getColumnValue(index) {
+                return this.columns[index].value;
+            }
+
+            dataListener = (x0, y0, x1, y1) => {
+                const data = this._data.slice(x0, x1).map((col) => col.slice(y0, y1));
+                const column_headers = this._columnHeaders.slice(x0, x1);
+                const num_columns = this._data.length;
+                const num_rows = this._data[0].length;
+                return {
+                    num_rows,
+                    num_columns,
+                    column_headers,
+                    data,
+                };
+            };
+        }
+    </script>
+    <script>
+
+        const baseColumns = ["Small", "Medium", "Large"];
+        const columnCount = 12;
+        const cellCount = 1000;
+        const columnSizes = new Map();
+
+        // Create data model that returns a dataListener function and a columns array.
+        const dataModel = new CreateDataModel(baseColumns, columnCount, cellCount);
+
+        // Get Regular Table element.
+        const tableApi = document.getElementsByTagName("regular-table")[0];
+
+        // Remove classes previously added by this api.
+        function removeClasses(element) {
+            element.classList.remove("small", "medium", "large");
+        }
+
+        // Add class to cell element according its name.
+        function setClass(element) {
+            const { x: index } = tableApi.getMeta(element);
+            const name = dataModel.getColumnKey(index);
+            let className;
+            if (name.includes("Small")) {
+                className = "small";
+            } else if (name.includes("Medium")) {
+                className = "medium";
+            } else if (name.includes("Large")) {
+                className = "large";
+            }
+            element.classList.add(className);
+        }
+
+        // Create a custom resizing element and append it to an element.
+        function setCustomResizeElement(th) {
+            const span = document.createElement("span");
+            span.classList.add("resize");
+            th.appendChild(span);
+        }
+
+        // Toggle a class to a cell according to whether its content is overflowed.
+        function toggleCellClip(el) {
+            el.classList.toggle("cell-clip", el.offsetWidth < el.scrollWidth);
+        }
+
+        // Provided an index, get column measures either by cache or computed styles.
+        function getColumnMeasures(element, index) {
+            const columnSize = columnSizes.get(index);
+            let width, minWidth, maxWidth;
+            if (columnSize) {
+                width = columnSize.width;
+                minWidth = columnSize.minWidth;
+                maxWidth = columnSize.maxWidth;
+            } else {
+                element.style.minWidth = "";
+                element.style.maxWidth = "";
```

Comment:
I will add a comment on this later on the code, but this two lines are crucial for this approach, since we need not to have inline styles applied in the element in order to get the values defined in the CSS rules when using the getComputedStyles api.